arelle/ModelXbrl.py

```diff
@@ -455,26 +379,26 @@ def matchSubstitutionGroup(self, elementQname, subsGrpMatchTable):
                 subsGrpMdlObj = subsGrpMdlObj.substitutionGroup
         return subsGrpMatchTable.get(None)
 
-    def isInSubstitutionGroup(self, elementQname, subsGrpQnames) -> bool:
+    def isInSubstitutionGroup(self, elementQname: QName, subsGrpQnames: QName | list[QName] | None) -> bool | None:
         """Determine if element is in substitution group(s)
-
         Used by ModelObjectFactory to return Class type for new ModelObject subclass creation, and isInSubstitutionGroup
 
         :param elementQname: Element/Concept QName to determine if in substitution group(s)
-        :type elementQname: QName
         :param subsGrpQnames: QName or list of QNames
-        :type subsGrpMatchTable: QName or [QName]
-        :returns: bool -- True if element is in any substitution group
         """
-        return self.matchSubstitutionGroup(elementQname, {
-                  qn:(qn is not None) for qn in (subsGrpQnames if hasattr(subsGrpQnames, '__iter__') else (subsGrpQnames,)) + (None,)})
 
-    def createInstance(self, url=None):
-        """Creates an instance document for a DTS which didn't have an instance document, such as
+        if isinstance(subsGrpQnames, list):
+            qnames = subsGrpQnames
+        else:
+            qnames = [cast('QName', subsGrpQnames)]
+        return cast('Optional[bool]', self.matchSubstitutionGroup(elementQname, {
+                  qn: (qn is not None) for qn in qnames}))
```

Comment:
This is also a breaking change (causing table linkbase conformance suite failures). Previously this function would handle any iterable (`hasattr(subsGrpQnames, '__iter__')`), but now only accommodates lists. Additionally, the `+ (None,)` bit was gross, but important because `matchSubstitutionGroup()` falls back to looking up `None`. A better approach would be to explicitly handle `None` here.

```suggestion
        from collections.abc import Iterable
        if isinstance(subsGrpQnames, Iterable):
            qnames = subsGrpQnames
        else:
            qnames = [subsGrpQnames]
        matchingSubstitutionGroup = cast('Optional[bool]', self.matchSubstitutionGroup(elementQname, {
                  qn: (qn is not None) for qn in qnames}))
        return matchingSubstitutionGroup if matchingSubstitutionGroup is not None else False
```